src/Cache/Cache.php

```diff
@@ -94,13 +94,13 @@ public function getOrSet(string $key, $ttl, callable $function, array $arguments
      *
      * @param string $className
      * @param string $functionName
-     * @param \string[] ...$uniqueValues
+     * @param \null[]|\string[] ...$uniqueValues
      * @return string
      */
-    public function keyHelper(string $className, string $functionName, string ...$uniqueValues): string
+    public function keyHelper(string $className, string $functionName, ?string ...$uniqueValues): string
     {
         // Please help prevent cache namespace collisions by driving carefully
-        $uniqueValues = str_replace('.', '_', $uniqueValues);
+        $uniqueValues = str_replace('.', '_', array_filter($uniqueValues));
```

Comment:
Yep, I think that'd result in `foo_1` and `foo__1` in the above two examples. It's a bit opaque though, is there some other magic character that we could use instead? This list of valid cache keys is pretty small though :(